tasks/git-hooks/govet.py

```diff
@@ -1,44 +1,28 @@
 #!/usr/bin/env python3
 
-import os.path
 import subprocess
 import sys
-
-# Exclude some folders since go vet fails there
-EXCLUDED_FOLDERS = {
-    "./cmd/agent/android",
-    "./cmd/agent/windows/service",
-    "./cmd/cluster-agent",
-    "./cmd/cluster-agent/app",
-    "./cmd/systray",
-    "./pkg/clusteragent/orchestrator",
-    "./pkg/process/config/testdata",
-    "./pkg/process/util/orchestrator",
-    "./pkg/trace/test/testsuite/testdata",
-    "./pkg/util/containerd",
-    "./pkg/util/containers/cri/crimock",
-    "./pkg/util/containers/providers/cgroup",
-    "./pkg/util/containers/providers/windows",
-    "./pkg/util/hostname/apiserver",
```

Comment:
per the comment above, go vet may fail on this folders, we should check if this is no longer the case

Reply:
CI runs `go vet` over all modules, so ✅ below shows it's OK :)

The comment is about go vet not vetting a package that isn't in the current module, which is what this PR fixes.

Reply:
CI runs `go vet` with a different set of build tags; the current pre-commit will fail:

<details>

<summary>`pre-commit run govet -a` output</summary>

```
❯ pre-commit run govet -a
govet....................................................................Failed
- hook id: govet
- exit code: 255

vet {'./pkg/aggregator/ckey/tests', './pkg/security/agent', './pkg/serverless/trace', './cmd/manager', './pkg/collector/corechecks/snmp/common', './pkg/config/remote/service/meta', './pkg/serializer/split', './pkg/collector/corechecks/cluster/ksm', './pkg/trace/traceutil', './cmd/security-agent/api', './pkg/util/cachedfetch', './pkg/trace/agent', './pkg/collector/corechecks', './cmd/process-agent/api', './pkg/runtime', './pkg/security/config', './pkg/util/containers/cri/crimock', './pkg/metadata/resources', './pkg/util/cloudproviders/gce', './pkg/clusteragent/externalmetrics', './pkg/snmp/traps', './cmd/agent/api', './pkg/util/containers', './pkg/workloadmeta/collectors/kubemetadata', './pkg/telemetry/utils', './pkg/logs/input/docker', './pkg/collector/corechecks/system/filehandles', './pkg/clusteragent/admission', './pkg/network/http/testutil', './pkg/secrets/test/input', './pkg/logs/metrics', './pkg/util/http', './pkg/util/androidasset', './pkg/util/kubernetes/clustername', './cmd/security-agent', './pkg/util/input', './pkg/clusteragent/custommetrics', './pkg/trace/watchdog', './pkg/collector/corechecks/system/cpu', './pkg/util/system', './pkg/logs/pb', './pkg/util/flavor', './pkg/compliance/mocks', './pkg/ebpf/bytecode/runtime', './pkg/autodiscovery/common/types', './pkg/util/kubernetes', './pkg/metadata/common', './pkg/network/tracer/connection/kprobe', './cmd/agent/api/response', './pkg/autodiscovery/providers', './pkg/workloadmeta/collectors/ecsfargate', './pkg/collector/check/defaults', './pkg/process/net/resolver', './cmd/security-agent/common', './pkg/collector/runner/tracker', './cmd/system-probe', './pkg/security/probe/doc_generator', './pkg/serverless/logs', './pkg/flare', './pkg/serializer/marshaler', './pkg/process/dockerproxy', './pkg/util/kernel', './pkg/trace/config/features', './pkg/autodiscovery/configresolver', './pkg/status/health', './pkg/collector/corechecks/containers/generic', './pkg/workloadmeta', './pkg/util/ecs/metadata', './pkg/network/filter', './pkg/serverless/flush', './pkg/logs/status', './pkg/logs/diagnostic', './pkg/serverless/registration', './cmd/cluster-agent/commands', './pkg/process/config', './pkg/collector/loaders', './pkg/trace/test/testutil', './pkg/otlp/internal/testutil', './pkg/logs/auditor', './pkg/util/containers/providers/mock', './cmd/agent/common/misconfig', './pkg/telemetry', './pkg/util/docker/fake', './cmd/agent/api/agent', './pkg/trace/writer', './pkg/clusteragent/admission/controllers/webhook', './pkg/network/config', './pkg/errors', './pkg/orchestrator/redact', './pkg/api/healthprobe', './pkg/forwarder/endpoints', './pkg/network/nettop', './pkg/api/util', './pkg/collector/corechecks/containers', './pkg/logs/tag', './pkg/network/driver', './cmd/system-probe/config', './pkg/clusteragent/orchestrator', './pkg/secrets/test/argument', './pkg/logs/service', './pkg/util/xc', './pkg/security/tests', './pkg/metadata/inventories', './pkg/proto/utils', './pkg/trace/api/apiutil', './pkg/network/ebpf', './pkg/workloadmeta/collectors/kubelet', './pkg/logs/input/listener', './pkg/trace/test', './pkg/collector/corechecks/snmp/report', './pkg/network/encoding', './pkg/util/kubernetes/apiserver/leaderelection', './pkg/util/hostname/validate', './pkg/process/util/api', './pkg/config/resolver', './pkg/util/containerd', './cmd/cluster-agent/api/agent', './pkg/util/kubernetes/certificate', './pkg/collector/corechecks/snmp/session', './pkg/collector/corechecks/systemd', './pkg/util/json', './pkg/tagger/replay', './pkg/aggregator', './pkg/aggregator/ckey', './pkg/collector/corechecks/containers/containerd', './pkg/otlp/internal/serializerexporter', './pkg/collector/corechecks/snmp/discovery', './pkg/util/cache', './pkg/trace/metrics/timing', './pkg/collector/corechecks/embed/jmx', './pkg/network/tracer/testutil', './pkg/compliance/agent', './pkg/trace/sampler', './pkg/config/settings', './pkg/logs/pipeline', './pkg/autodiscovery/scheduler', './pkg/tagger/subscriber', './pkg/workloadmeta/collectors/containerd', './pkg/jmxfetch', './pkg/collector/corechecks/snmp/devicecheck', './pkg/config/remote/service/tuf', './pkg/clusteragent/custommetrics/api/generated/openapi', './pkg/security/tests/syscall_tester', './pkg/trace/test/testsuite', './pkg/process/statsd', './pkg/serverless/tags', './pkg/clusteragent/clusterchecks', './cmd/systray', './pkg/util/testutil', './pkg/collector/python', './pkg/util/hostname/kubelet', './pkg/network/http/debugging', './pkg/tagger/types', './pkg/clusteragent/clusterchecks/types', './pkg/network/netlink/testutil', './cmd/iot-agent', './cmd/cluster-agent/admission', './pkg/logs/input/kubernetes', './pkg/collector/corechecks/containers/docker', './pkg/util/kubernetes/tags', './pkg/dogstatsd/replay', './pkg/collector/corechecks/nvidia/jetson', './pkg/diagnose', './pkg/logs/pipeline/mock', './pkg/collector/corechecks/net', './pkg/compliance/checks/env', './pkg/workloadmeta/collectors', './pkg/collector/corechecks/ebpf/probe', './pkg/workloadmeta/collectors/docker', './pkg/process/encoding/request', './pkg/util/common', './pkg/util/jsonquery', './pkg/util/containers/v2/metrics/system', './pkg/trace/atomic', './pkg/collector/corechecks/snmp', './pkg/config/remote/store', './pkg/trace/logutil', './pkg/config/seelog', './cmd/cluster-agent/app', './cmd/agent/windows/service', './pkg/util/retry', './pkg/util/fargate', './pkg/util', './cmd/agent/common/commands', './cmd/trace-agent', './pkg/util/grpc', './cmd/cluster-agent-cloudfoundry', './cmd/cluster-agent', './pkg/network/dns', './pkg/forwarder/internal/retry', './pkg/tagger', './pkg/process/procutil', './pkg/trace/stats', './pkg/security/log', './pkg/util/containers/v2/metrics/docker', './pkg/logs/client/mock', './pkg/logs/scheduler', './pkg/util/docker', './pkg/logs', './pkg/compliance/checks', './pkg/tagger/telemetry', './pkg/process/net', './pkg/forwarder', './pkg/clusteragent/custommetrics/api', './pkg/workloadmeta/testing', './cmd/agent/common/signals', './pkg/logs/restart', './pkg/collector/metadata/agentchecks', './pkg/proto/pbgo/mocks', './pkg/serverless', './pkg/util/compression', './pkg/security/probe', './pkg/util/cloudproviders/alibaba', './pkg/clusteragent', './cmd/system-probe/utils', './pkg/pidfile', './pkg/ebpf/compiler', './pkg/collector/corechecks/snmp/valuestore', './pkg/secrets/test/simple', './pkg/collector/corechecks/cluster/kubernetesapiserver', './pkg/tagger/collectors', './pkg/secrets', './pkg/util/containers/v2/metrics/containerd', './cmd/secrets', './pkg/logs/input', './cmd/system-probe/app', './cmd/agent', './pkg/version', './pkg/tagger/remote', './pkg/network/ebpf/probes', './pkg/security/module', './pkg/autodiscovery/common/utils', './pkg/trace/filters', './pkg/util/ecs/metadata/v2', './pkg/util/containers/v2/metrics', './pkg/logs/decoder', './pkg/dogstatsd/listeners', './pkg/secrets/test/error', './pkg/logs/parser', './pkg/collector/corechecks/snmp/metadata', './pkg/process/encoding', './cmd/agent/android', './pkg/clusteragent/externalmetrics/model', './pkg/collector/corechecks/system', './pkg/logs/client', './pkg/serializer', './pkg/compliance/checks/custom', './pkg/network', './pkg/metadata', './pkg/autodiscovery/integration', './pkg/util/containers/providers/cgroup', './pkg/metadata/gohai', './pkg/util/hostname/file', './cmd/agent/common', './pkg/collector/corechecks/snmp/gosnmplib', './cmd/system-probe/windows/service', './pkg/process/util', './pkg/metadata/externalhost', './pkg/trace/osutil', './pkg/network/http', './cmd/dogstatsd', './pkg/util/containers/metrics', './pkg/collector', './pkg/clusteragent/api/v1', './pkg/trace/flags', './pkg/orchestrator/config', './pkg/logs/client/tcp', './pkg/tagger/utils', './pkg/logs/message', './pkg/util/containers/v2/metrics/provider', './pkg/trace/obfuscate', './pkg/tagset', './pkg/ebpf/bytecode/bindata', './pkg/util/executable', './pkg/util/hostname/apiserver', './pkg/util/kubernetes/apiserver/leaderelection/metrics', './pkg/util/profiling', './pkg/ebpf', './pkg/util/kubernetes/hostinfo', './pkg/collector/corechecks/system/disk', './pkg/util/kubernetes/autoscalers', './pkg/config/autodiscovery', './pkg/snmp', './pkg/config', './pkg/util/clusteragent', './cmd/agent/app/settings', './pkg/serializer/stream', './pkg/epforwarder', './pkg/logs/input/file', './pkg/diagnose/diagnosis', './pkg/network/netlink', './pkg/serverless/proc', './pkg/collector/corechecks/snmp/checkconfig', './pkg/logs/processor', './pkg/process/util/api/config', './pkg/collector/runner/expvars', './pkg/util/cloudproviders/tencent', './pkg/autodiscovery', './pkg/process/config/testdata', './pkg/util/ecs', './pkg/util/containers/v2/metrics/ecsfargate', './pkg/dogstatsd/packets', './pkg/config/legacy', './pkg/logs/client/http', './pkg/util/ecs/common', './pkg/config/remote/service', './pkg/util/containers/providers', './pkg/persistentcache', './pkg/collector/check', './pkg/config/settings/http', './pkg/util/kubernetes/apiserver', './pkg/security/ebpf/kernel', './pkg/collector/corechecks/system/winproc', './pkg/util/orchestrator', './cmd/agent/api/check', './pkg/logs/input/traps', './pkg/collector/scheduler', './cmd/cluster-agent-cloudfoundry/app', './pkg/collector/corechecks/containers/cri', './pkg/autodiscovery/listeners', './pkg/util/cloudproviders/cloudfoundry', './pkg/compliance/eval', './cmd/serverless', './pkg/collector/corechecks/system/uptime', './cmd/security-agent/api/agent', './pkg/util/hostname', './pkg/collector/corechecks/snmp/fetch', './pkg/security/ebpf/probes', './pkg/collector/corechecks/system/memory', './pkg/clusteragent/admission/controllers/secret', './pkg/collector/worker', './pkg/kubestatemetrics/store', './cmd/cluster-agent/api/v1', './cmd/agent/app', './pkg/metadata/host', './pkg/trace/api', './pkg/trace/pb/otlppb', './pkg/workloadmeta/collectors/ecs', './pkg/util/kubernetes/kubelet', './pkg/network/testutil', './pkg/forwarder/transaction', './pkg/collector/corechecks/cluster/orchestrator', './pkg/api/security', './cmd/process-agent/flags', './pkg/trace/metrics', './pkg/logs/input/container', './pkg/dogstatsd', './pkg/logs/input/journald', './pkg/util/cloudproviders', './cmd/agent/clcrunnerapi/v1', './pkg/network/tracer/connection', './pkg/util/ec2', './pkg/util/cgroups', './pkg/secrets/test/timeout', './pkg/util/ecs/metadata/testutil', './pkg/collector/corechecks/embed', './pkg/clusteragent/admission/metrics', './pkg/metadata/host/container', './pkg/orchestrator', './pkg/metrics', './cmd/system-probe/modules', './pkg/util/kubernetes/apiserver/common', './pkg/logs/input/windowsevent', './pkg/trace/pb', './pkg/util/ecs/metadata/v1', './pkg/secrets/test/response_too_long', './pkg/status', './pkg/network/config/sysctl', './pkg/workloadmeta/collectors/util', './pkg/trace/config', './pkg/logs/config', './pkg/serverless/daemon', './pkg/util/containers/collectors', './pkg/security/api', './pkg/clusteragent/admission/mutate', './pkg/util/cloudproviders/azure', './pkg/util/ecs/metadata/v3', './cmd/system-probe/api', './pkg/util/filesystem', './pkg/kubestatemetrics/builder', './pkg/logs/input/channel', './pkg/security/utils', './cmd/agent/app/standalone', './pkg/collector/runner', './pkg/metadata/v5', './pkg/logs/auditor/mock', './pkg/appsec', './pkg/util/containers/providers/windows', './pkg/logs/sender', './pkg/ebpf/bytecode', './pkg/trace/event', './pkg/process/util/api/headers', './pkg/process/checks', './pkg/autodiscovery/providers/names', './cmd/agent/clcrunnerapi', './cmd/cluster-agent/api', './pkg/clusteragent/admission/common', './pkg/security/metrics', './pkg/tagger/tagstore', './pkg/compliance', './pkg/serverless/metrics', './pkg/otlp', './cmd/process-agent', './pkg/aggregator/mocksender', './pkg/compliance/event', './pkg/proto/pbgo', './pkg/collector/corechecks/ebpf', './pkg/trace/test/testsuite/testdata', './pkg/trace/info', './pkg/collector/corechecks/cluster', './cmd/py-launcher', './pkg/security/ebpf', './pkg/tagger/local', './pkg/util/tmplvar', './cmd/system-probe/api/module', './pkg/dogstatsd/mapper', './pkg/security/probe/syscall_table_generator', './cmd/security-agent/app', './pkg/util/containers/cri', './cmd/secrets/providers', './pkg/network/tracer', './pkg/util/backoff', './cmd/agent/gui', './pkg/collector/metadata', './cmd/cluster-agent/custommetrics', './cmd/agent/common/commands/config'} in .
package github.com/DataDog/datadog-agent/pkg/util/containers/cri/crimock: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/pkg/util/containers/cri/crimock
package github.com/DataDog/datadog-agent/pkg/clusteragent/externalmetrics: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/pkg/clusteragent/externalmetrics
package github.com/DataDog/datadog-agent/pkg/ebpf/bytecode/runtime: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/pkg/ebpf/bytecode/runtime
package github.com/DataDog/datadog-agent/pkg/network/tracer/connection/kprobe: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/pkg/network/tracer/connection/kprobe
package github.com/DataDog/datadog-agent/cmd/system-probe: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/cmd/system-probe
package github.com/DataDog/datadog-agent/pkg/security/probe/doc_generator: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/pkg/security/probe/doc_generator
package github.com/DataDog/datadog-agent/pkg/network/filter: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/pkg/network/filter
package github.com/DataDog/datadog-agent/cmd/cluster-agent/commands: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/cmd/cluster-agent/commands
package github.com/DataDog/datadog-agent/pkg/util/docker/fake: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/pkg/util/docker/fake
package github.com/DataDog/datadog-agent/pkg/clusteragent/admission/controllers/webhook: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/pkg/clusteragent/admission/controllers/webhook
package github.com/DataDog/datadog-agent/pkg/network/driver: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/pkg/network/driver
package github.com/DataDog/datadog-agent/pkg/network/ebpf: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/pkg/network/ebpf
package github.com/DataDog/datadog-agent/pkg/util/kubernetes/apiserver/leaderelection: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/pkg/util/kubernetes/apiserver/leaderelection
package github.com/DataDog/datadog-agent/pkg/util/containerd: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/pkg/util/containerd
package github.com/DataDog/datadog-agent/pkg/util/kubernetes/certificate: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/pkg/util/kubernetes/certificate
package github.com/DataDog/datadog-agent/pkg/network/tracer/testutil: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/pkg/network/tracer/testutil
package github.com/DataDog/datadog-agent/pkg/security/tests/syscall_tester: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/pkg/security/tests/syscall_tester
package github.com/DataDog/datadog-agent/cmd/systray: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/cmd/systray
package github.com/DataDog/datadog-agent/pkg/collector/python: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/pkg/collector/python
package github.com/DataDog/datadog-agent/pkg/network/netlink/testutil: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/pkg/network/netlink/testutil
package github.com/DataDog/datadog-agent/cmd/cluster-agent/admission: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/cmd/cluster-agent/admission
package github.com/DataDog/datadog-agent/pkg/util/kubernetes/tags: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/pkg/util/kubernetes/tags
package github.com/DataDog/datadog-agent/cmd/agent/windows/service: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/cmd/agent/windows/service
package github.com/DataDog/datadog-agent/cmd/cluster-agent-cloudfoundry: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/cmd/cluster-agent-cloudfoundry
package github.com/DataDog/datadog-agent/cmd/cluster-agent: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/cmd/cluster-agent
package github.com/DataDog/datadog-agent/pkg/clusteragent/custommetrics/api: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/pkg/clusteragent/custommetrics/api
package github.com/DataDog/datadog-agent/pkg/security/probe: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/pkg/security/probe
package github.com/DataDog/datadog-agent/pkg/ebpf/compiler: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/pkg/ebpf/compiler
package github.com/DataDog/datadog-agent/cmd/secrets: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/cmd/secrets
package github.com/DataDog/datadog-agent/pkg/network/ebpf/probes: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/pkg/network/ebpf/probes
package github.com/DataDog/datadog-agent/cmd/agent/android: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/cmd/agent/android
package github.com/DataDog/datadog-agent/pkg/clusteragent/externalmetrics/model: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/pkg/clusteragent/externalmetrics/model
package github.com/DataDog/datadog-agent/pkg/util/containers/providers/cgroup: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/pkg/util/containers/providers/cgroup
package github.com/DataDog/datadog-agent/cmd/system-probe/windows/service: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/cmd/system-probe/windows/service
package github.com/DataDog/datadog-agent/pkg/util/hostname/apiserver: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/pkg/util/hostname/apiserver
package github.com/DataDog/datadog-agent/pkg/util/kubernetes/autoscalers: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/pkg/util/kubernetes/autoscalers
package github.com/DataDog/datadog-agent/pkg/security/ebpf/kernel: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/pkg/security/ebpf/kernel
package github.com/DataDog/datadog-agent/pkg/util/orchestrator: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/pkg/util/orchestrator
package github.com/DataDog/datadog-agent/cmd/cluster-agent-cloudfoundry/app: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/cmd/cluster-agent-cloudfoundry/app
package github.com/DataDog/datadog-agent/pkg/security/ebpf/probes: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/pkg/security/ebpf/probes
package github.com/DataDog/datadog-agent/pkg/clusteragent/admission/controllers/secret: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/pkg/clusteragent/admission/controllers/secret
package github.com/DataDog/datadog-agent/pkg/kubestatemetrics/store: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/pkg/kubestatemetrics/store
package github.com/DataDog/datadog-agent/pkg/util/cgroups: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/pkg/util/cgroups
package github.com/DataDog/datadog-agent/pkg/clusteragent/admission/metrics: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/pkg/clusteragent/admission/metrics
package github.com/DataDog/datadog-agent/pkg/util/kubernetes/apiserver/common: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/pkg/util/kubernetes/apiserver/common
package github.com/DataDog/datadog-agent/pkg/clusteragent/admission/mutate: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/pkg/clusteragent/admission/mutate
package github.com/DataDog/datadog-agent/pkg/util/containers/providers/windows: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/pkg/util/containers/providers/windows
package github.com/DataDog/datadog-agent/pkg/clusteragent/admission/common: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/pkg/clusteragent/admission/common
package github.com/DataDog/datadog-agent/cmd/py-launcher: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/cmd/py-launcher
package github.com/DataDog/datadog-agent/pkg/security/ebpf: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/pkg/security/ebpf
package github.com/DataDog/datadog-agent/cmd/secrets/providers: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/cmd/secrets/providers
package github.com/DataDog/datadog-agent/cmd/cluster-agent/custommetrics: build constraints exclude all Go files in /Users/$USER/go/src/github.com/DataDog/datadog-agent/cmd/cluster-agent/custommetrics
```

</details>

I guess the ideal solution here is to use the same set of build tags we use on the CI?

Reply:
Ah, I have those build tags enabled by default (so gopls works) so I didn't see that.

I feel like I've worked around this "feature" in Go so many times!  Maybe easiest to just add an empty "stub.go" to each package?

Reply:
> Maybe easiest to just add an empty "stub.go" to each package?
Sure, that should also work

Reply:
Hm, that gets down to just one error:
```
tereshkova ~/go/src/github.com/DataDog/datadog-agent [dustin.mitchell/ac-1370*] $ pre-commit run govet -a
govet....................................................................Failed
- hook id: govet
- exit code: 255

package github.com/DataDog/datadog-agent/pkg/ebpf/compiler: C++ source files not allowed when not using cgo or SWIG: compiler.cpp files.cpp shim.cpp
```
That doesn't appear to be a `go vet` analyzer error, but rather a go error.  Why does it think cgo is not enabled?  Adding CGO_ENABLED=1 doesn't change the behavior.  Any ideas?

Reply:
Doing [this](https://stackoverflow.com/a/30270409) changed the error for me to
```
❯ go vet ./pkg/ebpf/compiler
# github.com/DataDog/datadog-agent/pkg/ebpf/compiler
In file included from compiler.cpp:1:
pkg/ebpf/compiler/compiler.h:5:10: fatal error: 'clang/Driver/Compilation.h' file not found
```
(I am not sure if that helps but I don't have better ideas)

Reply:
You won't be able to compile the `compiler` package unless you are on Linux and have our custom clang dependencies available. See https://github.com/DataDog/datadog-agent/blob/07f4df2f4ab7b2120150782e8931e6f287632de5/.gitlab/binary_build/system_probe.yml#L14

Reply:
It looks like we should exclude this one specific package from vetting, then.

Reply:
I don't love the stub file solution. Is there a way with `go list` to get a list of valid packages, given a set of build tags?

Reply:
I suppose we could iterate over all of the packages and run `go list` on them -- it does fail with a nonzero exit status on packages with all files excluded.

Reply:
Hm, so I tried `go list {pkg}` for each package in each module.  It's incredibly slow.

I tried `go list ./...` for each module, which only returns valid packages, and then excluding any invalid packages.  That's really fiddly with all of the required path manipulation (`go list` returns GOPATH-qualified paths, but we need modules and module-relative package paths, which have to have `./`, etc.).

It fails with:
```
# github.com/DataDog/datadog-agent/pkg/collector/python
_cgo_export.c: In function ‘_cgo_b5c4a8467f4b_Cfunc__Cmalloc’:
_cgo_export.c:539:2: warning: ‘malloc’ is deprecated: raw primitives should not be used in the contextof the rtloader [-Wdeprecated-declarations]
  539 |  ret = malloc(a->p0);
      |  ^~~
In file included from _cgo_export.c:3:
/usr/include/stdlib.h:539:14: note: declared here
  539 | extern void *malloc (size_t __size) __THROW __attribute_malloc__
      |              ^~~~~~
_cgo_export.c:541:3: warning: ‘malloc’ is deprecated: raw primitives should not be used in the contextof the rtloader [-Wdeprecated-declarations]
  541 |   ret = malloc(1);
      |   ^~~
In file included from _cgo_export.c:3:
/usr/include/stdlib.h:539:14: note: declared here
  539 | extern void *malloc (size_t __size) __THROW __attribute_malloc__
      |              ^~~~~~
# github.com/DataDog/datadog-agent/cmd/py-launcher
cmd/py-launcher/py-launcher.go:22:10: fatal error: datadog_agent_rtloader.h: No such file or directory
   22 | #include "datadog_agent_rtloader.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```
even if those two packages (./cmd/py-launcher and ./pkg/collector/python) aren't included on the `go vet` command line.  I'll make a push with this change.

Reply:
`go list` is actually super powerful in terms of outputting what you want. It defaults to `{{ .ImportPath }}`, but you can specify an alternate template. See https://pkg.go.dev/cmd/go/internal/list for details.

Thanks for giving `go list` a shot. I think this will be a better solution in the long run.

Reply:
Perhaps.  But, even with that sorted, I'm still stuck on the compilation issues.

Reply:
Apologies, I missed that the failures were running with `go list`. `go list` does take build flags. Perhaps we need to use the same ones that are used for `inv agent.build`?